### PR TITLE
Add AnimeHeaven presence

### PR DESCRIPTION
**Pull request by marceau-maubert**
Hopefully this works out a little better.

# Screenshots

Screenshot_2021-03-21 AnimeHeaven - Watch HD Anime Online Free

Screenshot_2021-03-21 Sorcery Fight Full Episodes Online Free AnimeHeaven

Screenshot_2021-03-21 Watch Sorcery Fight Episode 23 Online Free AnimeHeaven

### CODE REVIEW

File: websites/A/AnimeHeaven/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
+  "service": "AnimeHeaven",
+  "author": {
+    "name": "Tenri",
+    "id": "1999"
```
Comment (SlowLife1661) [inline websites/A/AnimeHeaven/dist/metadata.json:6]: Didn't know snowflake like this exist

Comment (marceau-maubert) [reply websites/A/AnimeHeaven/dist/metadata.json:6]: Sorry, thought it was a discriminator I needed to put in. I'll fix that

**Comment by ririxidev** [comment]
Heya @marceau-maubert! We have some news for you that you should take look at:
We are so sorry but we need to deny your pull request regarding this presence due of (will repeat again)
 - Your presence is low quality coded.
 - There isn't a lot of subpages added for support.
 
Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Comment by marceau-maubert** [comment]
- Again, compare my presence to [this one](https://github.com/PreMiD/Presences/blob/6e20915d62d680691ed2322562a4ff0c76b4ed98/websites/T/twist.moe/presence.ts), which *was added by [a member of the PreMid organisation](https://github.com/TheDropX)*, what makes my presence "low quality coded", considering I'm basing mine off of it?
- Browse the website I'm trying to support for more than 5 minutes and tell me honestly if you think there's any "subpages" that actually need to be implemented?? It's not any more complex than the presence I linked above.

You're not being any helpful.

**Comment by ririxidev** [comment]
Hey @marceau-maubert again,
> Again, compare my presence to this one, which was added by a member of the PreMid organisation, what makes my presence "low quality coded", considering I'm basing mine off of it?
That's right but in this time it was old time ago and we didn't had stricted code style etc and quality of them.
> Browse the website I'm trying to support for more than 5 minutes and tell me honestly if you think there's any "subpages" that actually need to be implemented?? It's not any more complex than the presence I linked above.
image
Hope, that will help.

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Comment by marceau-maubert** [comment]
> That's right but in this time it was old time ago and we didn't had stricted code style etc and quality of them.

Okay, maybe, but you're still not telling me what part of my code is "low quality". I'm grabbing elements on the page and setting presence data, do you want any more sprinkles with that?

> Hope, that will help.

You want me to specifically mention on the webpage if I'm browsing each category... Why? That's really not the point.

**Comment by ririxidev** [comment]
Hey @marceau-maubert again,
> Okay, maybe, but you're still not telling me what part of my code is "low quality". I'm grabbing elements on the page and setting presence data, do you want any more sprinkles with that?

Literally we are not going to specify what part of code is low quality because we are not here to do it. We have our own ways to check if it is low quality or not and we are just saying that as the reason.

> You want me to specifically mention on the presence if I'm browsing each category... Why? That's really not the point.

Yup, if it is possible. Yes.

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Comment by marceau-maubert** [comment]
> Literally we are not going to specify what part of code is low quality because we are not here to do it. We have our own ways to check if it is low quality or not and we are just saying that as the reason.

First off, why are you speaking as if more than one reviewer has seen this pull request so far?

Second, if you're not here to *review code*, why are you part of the *Reviewer Team*, which have tools implemented just for that purpose in GitHub, and then why do you say that you have your own ways to check if it's not your job? Seems to me like the automated checks are fine with it, I don't know what else you need.

> Yup, if it is possible. Yes.

Fine, I've cleaned up the code and added more paths to handle just for you.

Also, if you actually intend for me to improve on my pull request, why do you close it immediately? I've made changes by now, but unless you reopen it, I'm going to have to create a third one.

**Comment by marceau-maubert** [comment]
I've pushed some null checks and cleaned the code for the most part, if that helps.

**Comment by Earu** [comment]
Heya @ririxdev! We have some news for you that you should take look at:

This man here is trying to commit something actually cool, and the only answer he is given is that it’s “low quality coded”, why just no simply point out what’s wrong with the pull request using the GitHub review feature ? There can’t be any improvement if you don’t point out what’s wrong with it in the first place... 

Thanks,
Earu, developer.
Contact: N/A

**Pull request closed without merging**